src/zino/tasks/bgpstatemonitortask.py

```diff
@@ -94,6 +94,10 @@ async def run(self):
 
         bgp_info = self._fixup_ip_addresses(bgp_info=bgp_info)
 
+        # BGP peers is by default None, so to record the BGP peers, we have to initialize it
+        if not self.device_state.bgp_peers:
+            self.device_state.bgp_peers = dict()
+
```

Comment:
Actually, I would go the other way around and make `DeviceState.bgp_peers` non-optional with a default value of `{}`, similar to `DeviceState.ports`.  You save three lines of unnecessary init-code that may have to be repeated elsewhere when things change in the future.

Reply:
fixed